packaging/os/apt_repository.py

```diff
@@ -321,6 +321,21 @@ def remove_source(self, line):
         source = self._parse(line, raise_if_invalid_or_disabled=True)[2]
         self._remove_valid_source(source)
 
+    @property
+    def repos_urls(self):
+        repositories = []
+        for parsed_repos in self.files.values():
+            for parsed_repo in parsed_repos:
+                enabled = parsed_repo[1]
+                source_line = parsed_repo[3]
+
+                if not enabled:
+                    continue
+
```

Comment:
And in addition to that, need to move the _expand_ppa method from UbuntuSourcesList to SourcesList.